icu4c/source/common/rbbidata.h

```diff
@@ -94,49 +95,54 @@ struct RBBIDataHeader {
 
 
 
-struct  RBBIStateTableRow {
-    int16_t          fAccepting;    /*  Non-zero if this row is for an accepting state.   */
+template <typename ST, typename UT>
+struct RBBIStateTableRowT {
+    ST               fAccepting;    /*  Non-zero if this row is for an accepting state.   */
                                     /*  Value 0: not an accepting state.                  */
                                     /*       -1: Unconditional Accepting state.           */
                                     /*    positive:  Look-ahead match has completed.      */
                                     /*           Actual boundary position happened earlier */
                                     /*           Value here == fLookAhead in earlier      */
                                     /*              state, at actual boundary pos.        */
-    int16_t          fLookAhead;    /*  Non-zero if this row is for a state that          */
+    ST               fLookAhead;    /*  Non-zero if this row is for a state that          */
```

Comment:
I think fLookAhead and fTagIdx could be unsigned - I don't think that they use negative values.

It would be nice if we didn't need to have two types as template parameters, but fAccepting needs -1 as a value. Maybe it would be possible to rearrange the code to dodge this and work with unsigned, but it's not trivial, so keeping the two types seems expedient.

Reply:
noted for follow-up